AppKit/CPTextView/CPTextView.j

```diff
@@ -0,0 +1,1890 @@
+{
+    [_caretTimer invalidate];
+    _caretTimer = nil;
+    _isFirstResponder = NO;
+    [self setNeedsDisplay:YES];
+
+    return YES;
+}
+
+
+#pragma mark -
+#pragma mark Delegate methods
+
+/*!
+    TODO : documentation
+*/
+- (void)setDelegate:(id <CPTextViewDelegate>)aDelegate
+{
+    if (aDelegate === _delegate)
+        return;
+
+    var notificationCenter = [CPNotificationCenter defaultCenter];
+
+    _delegateRespondsToSelectorMask = 0;
+
+    if (_delegate)
+    {
+        [notificationCenter removeObserver:_delegate name:CPTextDidChangeNotification object:self];
+        [notificationCenter removeObserver:_delegate name:CPTextViewDidChangeSelectionNotification object:self];
+        [notificationCenter removeObserver:_delegate name:CPTextViewDidChangeTypingAttributesNotification object:self];
+    }
+
+    _delegate = aDelegate;
+
+    if (_delegate)
+    {
+        if ([_delegate respondsToSelector:@selector(textDidChange:)])
+            [notificationCenter addObserver:_delegate selector:@selector(textDidChange:) name:CPTextDidChangeNotification object:self];
+
+        if ([_delegate respondsToSelector:@selector(textViewDidChangeSelection:)])
+            [notificationCenter addObserver:_delegate selector:@selector(textViewDidChangeSelection:) name:CPTextViewDidChangeSelectionNotification object:self];
+
+        if ([_delegate respondsToSelector:@selector(textViewDidChangeTypingAttributes:)])
+            [notificationCenter addObserver:_delegate selector:@selector(textViewDidChangeTypingAttributes:) name:CPTextViewDidChangeTypingAttributesNotification object:self];
+
+        if ([_delegate respondsToSelector:@selector(textView:doCommandBySelector:)])
+            _delegateRespondsToSelectorMask |= kDelegateRespondsTo_textView_doCommandBySelector;
+
+        if ([_delegate respondsToSelector:@selector(textShouldBeginEditing:)])
+            _delegateRespondsToSelectorMask |= kDelegateRespondsTo_textShouldBeginEditing;
+
+        if ([_delegate respondsToSelector:@selector(textView:willChangeSelectionFromCharacterRange:toCharacterRange:)])
+            _delegateRespondsToSelectorMask |= kDelegateRespondsTo_textView_willChangeSelectionFromCharacterRange_toCharacterRange;
+
+        if ([_delegate respondsToSelector:@selector(textView:shouldChangeTextInRange:replacementString:)])
+            _delegateRespondsToSelectorMask |= kDelegateRespondsTo_textView_shouldChangeTextInRange_replacementString;
+
+        if ([_delegate respondsToSelector:@selector(textView:shouldChangeTypingAttributes:toAttributes:)])
+            _delegateRespondsToSelectorMask |= kDelegateRespondsTo_textView_shouldChangeTypingAttributes_toAttributes;
+    }
+}
+
+
+#pragma mark -
+#pragma mark Key window methods
+
+- (void)becomeKeyWindow
+{
+    [self setNeedsDisplay:YES];
+}
+
+/*!
+    @ignore
+*/
+- (void)resignKeyWindow
+{
+    [self setNeedsDisplay:YES];
+}
+
+- (BOOL)_isFocused
+{
+   return [[self window] isKeyWindow] && _isFirstResponder;
```

Comment:
You should replace _isFirstResponder by `[[self window] firstResponder] === self ` and then remove this ivars